azure-pipelines-regression-test.yml

```diff
@@ -0,0 +1,309 @@
+# with the freshly built compiler from this repository.
+#
+# Purpose:
+# - Catch regressions early by testing the compiler against real-world F# libraries
+# - Provide detailed build logs when third-party builds fail with the new compiler
+# - Support extensible matrix configuration for testing multiple libraries
+#
+# Matrix Configuration:
+# The testMatrix parameter supports multiple repositories with their specific build requirements.
+# Each entry should specify:
+# - repo: GitHub repository in format "owner/name"
+# - commit: Specific commit SHA to test against for reproducible results
+# - buildScript: Build script to execute (relative to repo root)
+# - displayName: Human-readable name for the test job
+#
+# Current matrix includes:
+# - FSharpPlus: A popular F# library with extensive use of advanced F# features
+
+trigger:
+  branches:
+    include:
+    - main
+    - release/*
+    - feature/*
+  paths:
+    include:
+    - src/Compiler/*
+    - src/fsc/*
+    - src/FSharp.Core/*
+    - src/FSharp.Build/*
+    exclude:
+    - docs/*
+    - .github/*
+    - '*.md'
+
+pr:
+  branches:
+    include:
+    - main
+    - release/*
+    - feature/*
+  paths:
+    include:
+    - src/Compiler/*
+    - src/fsc/*
+    - src/FSharp.Core/*
+    - src/FSharp.Build/*
+    exclude:
+    - docs/*
+    - .github/*
+    - '*.md'
+
+variables:
+  - name: _TeamName
+    value: FSharp
+  - name: _BuildConfig
+    value: Release
+  - name: _PublishUsingPipelines
+    value: true
+  - name: Codeql.Enabled
+    value: false  # Disabled for regression tests
+  # Pick up pool provider name behavior from shared yaml template
+  - template: /eng/common/templates/variables/pool-providers.yml
+
+# Matrix of third-party repositories to test against
+# This configuration can be extended in the future
+parameters:
+- name: testMatrix
+  type: object
+  default:
+  - repo: fsprojects/FSharpPlus
+    commit: f614035b75922aba41ed6a36c2fc986a2171d2b8
+    buildScript: build.cmd
+    displayName: FSharpPlus
+
+stages:
+- stage: RegressionTest
+  displayName: F# Compiler Regression Tests
+  jobs:
+  - job: BuildCompiler
+    displayName: Build F# Compiler
+    pool:
+      name: $(DncEngPublicBuildPool)
+      demands: ImageOverride -equals windows.vs2022.amd64.open
+    timeoutInMinutes: 60
+    steps:
+    - checkout: self
+      clean: true
+      displayName: Checkout F# compiler source
+
+    - script: dotnet --list-sdks
+      displayName: Report .NET SDK versions
+
+    # We first download a publicly available .NET SDK. That one has support for `path` in global.json. 
+    # dotnet.cmd script can then download a version which is not yet shipped, but matches global.json.
+    - task: UseDotNet@2
+      displayName: install SDK
+      inputs:
+        packageType: sdk
+        version: '10.x'
+        includePreviewVersions: true
+        workingDirectory: $(Build.SourcesDirectory)
+        installationPath: $(Build.SourcesDirectory)/.dotnet
```

Comment:
Since you will be moving this section to just reference `EndToEndBuildTests`, this should be rather adapted to the repository being tested.

i.e. move a `UseDotNet` block to the testmatrix part. And for each repo being tested, it should respect that repository's own `global.json`  or use latest public .NET if there is no global.json.

In the default config, FsharpPlus, I can see the file https://github.com/fsprojects/FSharpPlus/blob/master/global.json .
The script should be clever enough to detect that and download correct SDK based on it (or default to latest public)